package/network/utils/iproute2/Makefile

```diff
@@ -127,12 +130,15 @@ define Build/Compile
 	+$(MAKE_VARS) $(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) $(MAKE_FLAGS)
 endef
 
+# use minimal functionality for dev libraries
+ifeq ($(BUILD_VARIANT),tiny)
```

Comment:
This will break user packages linking to libnetlink.a when ip-full is selected

Reply:
It seems you're not looking at the branch tip but reviewing outdated code 1 commit behind; please check again.

Reply:
@guidosarducci I think a lack of clarity from myself here.  Ideally each commit in the master tree should build without error.  Having a commit still in that 'breaks' stuff followed by a commit that fixes the break isn't ideal.  If you squash the fix commit into the corresponding prior commit so nothing 'breaks' on the way then I'm sure that would be better.

Reply:
This is simply the PR which you requested yourself and @ldir-EDB0 confirmed in https://github.com/openwrt/openwrt/pull/1627#issuecomment-465466958 and https://github.com/openwrt/openwrt/pull/1627#issuecomment-465489270 respectively:
>I'm fine opening a separate PR which re-applies the reverted commits together with the fixes

and
>I agree that a separate PR with the 4 reverted commits and the fix commit would be a fair way forward.

The branch tip **is** the "fix commit", and doesn't break iproute2 as you claim. The same commit could have been cleanly and quickly applied to master without a mass reversion, thus avoiding unneeded churn.